src/storage/scripting.cc

```diff
@@ -409,13 +384,42 @@ Status FunctionCall(redis::Connection *conn, const std::string &name, const std:
     std::string libcode;
     s = srv->FunctionGetCode(libname, &libcode);
     if (!s) return s;
-
     s = FunctionLoad(conn, libcode, false, false, &libname, read_only);
     if (!s) return s;
 
     lua_getglobal(lua, (REDIS_LUA_REGISTER_FUNC_PREFIX + name).c_str());
   }
 
+  uint64_t function_flags = read_only ? ScriptFlags::kScriptNoWrites : 0;
+  lua_getglobal(lua, (REDIS_LUA_REGISTER_FUNC_FLAGS_PREFIX + name).c_str());
+  if (!lua_isnil(lua, -1)) {
+    int n = static_cast<int>(lua_objlen(lua, -1));
+    for (int i = 1; i <= n; ++i) {
+      lua_pushnumber(lua, i);
+      lua_gettable(lua, -2);
+      std::string flag = lua_tostring(lua, -1);
+      if (flag == "no-writes") {
+        function_flags |= kScriptNoWrites;
+      } else if (flag == "allow-oom") {
+        return {Status::NotSupported, "allow-oom is not supported yet"};
+      } else if (flag == "allow-stale") {
+        return {Status::NotSupported, "allow-stale is not supported yet"};
+      } else if (flag == "no-cluster") {
+        function_flags |= kScriptNoCluster;
+      } else if (flag == "allow-cross-slot-keys") {
+        function_flags |= kScriptAllowCrossSlotKeys;
+      } else {
+        return {Status::NotOK, "Unexpected function flag: " + flag};
+      }
+      lua_pop(lua, 1);
+    }
+  }
+  lua_pop(lua, 1);
+
+  ScriptRunCtx script_run_ctx;
+  script_run_ctx.flags = function_flags;
+  SaveOnRegistry(lua, REGISTRY_SCRIPT_RUN_CTX_NAME, &script_run_ctx);
```

Comment:
I think we can parse these flags in function register phase instead of calling phase.

Reply:
also `SaveOnRegistry` seems not needed then.

Reply:
> also `SaveOnRegistry` seems not needed then.

If removed, how to obtain flags when the Lua script is executed?

Reply:
Don't forget to remove it after the function call is finished.